apm-agent-core/src/main/java/co/elastic/apm/agent/bci/ElasticApmAgent.java

```diff
@@ -647,7 +647,12 @@ public static synchronized void reset() {
         if (instrumentation == null) {
             return;
         }
-        GlobalTracer.get().require(Tracer.class).stop();
+        Tracer tracer = GlobalTracer.get().require(Tracer.class);
```

Comment:
we can use `assertionsEnabled` as a proxy for "this is running in tests" without having to do a cast, or even better do this transparently within `ElasticApmTracer#stop` without having the caller to deal with "is this test code or not".

Reply:
yes, you're right that's definitely better